### PR TITLE
fix: OSX build steps and README updates.

### PR DESCRIPTION
**Pull request by dougch**
*Issue #401 *

*Description:*

Build script for macOS and updates to the README for Mac to match the script.

Tests passing on OSX 10.14, cpp-sdk v1.7.163 and openssl@1.1 installed from brew.

```
PhaseScriptExecution CMake\ PostBuild\ Rules /var/tmp/build-20191004/build-csdk/aws-encryption-sdk.build/Debug/RUN_TESTS.build/Script-94C362D9F28F4E1F9AC4A310.sh
    cd /var/tmp/build-20191004/aws-encryption-sdk-c
    /bin/sh -c /var/tmp/build-20191004/build-csdk/aws-encryption-sdk.build/Debug/RUN_TESTS.build/Script-94C362D9F28F4E1F9AC4A310.sh
/usr/local/Cellar/cmake/3.15.2/bin/ctest --force-new-ctest-process -C Debug
Test project /private/var/tmp/build-20191004/build-csdk
      Start  1: t_cpputils.test
 1/21 Test  #1: t_cpputils.test ..................   Passed    1.12 sec
      Start  2: t_kms_keyring.test
 2/21 Test  #2: t_kms_keyring.test ...............   Passed    1.10 sec
      Start  3: cipher
 3/21 Test  #3: cipher ...........................   Passed    7.20 sec
      Start  4: header
 4/21 Test  #4: header ...........................   Passed    0.02 sec
      Start  5: materials
 5/21 Test  #5: materials ........................   Passed    0.02 sec
      Start  6: enc_ctx
 6/21 Test  #6: enc_ctx ..........................   Passed    0.04 sec
      Start  7: encrypt
 7/21 Test  #7: encrypt ..........................   Passed    0.93 sec
      Start  8: hkdf
 8/21 Test  #8: hkdf .............................   Passed    0.01 sec
      Start  9: raw_aes_keyring
 9/21 Test  #9: raw_aes_keyring ..................   Passed    0.01 sec
      Start 10: raw_rsa_keyring
10/21 Test #10: raw_rsa_keyring ..................   Passed    0.03 sec
      Start 11: multi_keyring
11/21 Test #11: multi_keyring ....................   Passed    0.01 sec
      Start 12: signature
12/21 Test #12: signature ........................   Passed    3.73 sec
      Start 13: trailing_sig
13/21 Test #13: trailing_sig .....................   Passed    0.02 sec
      Start 14: local_cache
14/21 Test #14: local_cache ......................   Passed    0.04 sec
      Start 15: caching_cmm
15/21 Test #15: caching_cmm ......................   Passed    0.04 sec
      Start 16: keyring_trace
16/21 Test #16: keyring_trace ....................   Passed    0.01 sec
      Start 17: decrypt_aes128_hkdf
17/21 Test #17: decrypt_aes128_hkdf ..............   Passed    1.07 sec
      Start 18: decrypt_aes128_hkdf_bad_header
18/21 Test #18: decrypt_aes128_hkdf_bad_header ...   Passed    0.01 sec
      Start 19: decrypt_hello
19/21 Test #19: decrypt_hello ....................   Passed    0.01 sec
      Start 20: decrypt_hello_tiny
20/21 Test #20: decrypt_hello_tiny ...............   Passed    0.01 sec
      Start 21: decryption_vectors
21/21 Test #21: decryption_vectors ...............   Passed   14.55 sec

100% tests passed, 0 tests failed out of 21

Total Test time (real) =  30.02 sec

** BUILD SUCCEEDED **
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -248,6 +248,15 @@ Build and install aws-c-common:
...
 You have successfully built and installed the AWS Encryption SDK for C.
 
+## Doxygen Documentation
+
+To build the documentation, you'll need [doxygen](http://www.doxygen.nl/) installed.  Check the [downloads](http://www.doxygen.nl/download.html) page 
+or use your OS package manager.
+
+Next, rerun the above cmake with a `-DBUILD_DOC="ON"` flag before building aws-encryption-sdk-c.
+
+Finally, run either `make doc_doxygen` (Unix) or `xcodebuild -scheme doc_doxygen` (Mac) to generate the documentation.
```
Comment (ttjsu-aws) [inline README.md]: Windows instructions for Doxygen is missing here.

File: codebuild/osx.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    echo "Can't find brew, required to build"
+    exit 1
+  fi
+  brew install ${OPENSSLVER} cmake || true
+  mkdir ${BUILDROOT} || true
+  mkdir ${INSTALLROOT} || true
+}
+
+build_cpp(){
+    echo "Building cpp"
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: `echo "Building aws-sdk-cpp"`

File: codebuild/osx.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    # See https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/cmake-params.html#cmake-target-arch
+    cmake -G Xcode -DTARGET_ARCH="APPLE" -DBUILD_SHARED_LIBS=ON -DBUILD_ONLY="kms" -DENABLE_UNITY_BUILD=ON ../aws-sdk-cpp 
+    # This target runs the cpp tests, but doesn't appear to run the other dependancy tests.
+    xcodebuild -target ALL_BUILD 
+    xcodebuild -target install
+
+}
+
+build_csdk(){
+    echo "Building csdk"
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: `echo "Building aws-encryption-sdk-c"`

File: codebuild/osx.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+
+set +ex
+
+OS=$(uname -s)
+CPP_VER="1.7.163"  #github versioned branch
+BUILDROOT="/var/tmp/build-$(date +%s)"
+OPENSSLVER="openssl@1.1" #note this is a brew label, not an exact version reference
+BUILDROOT="/var/tmp/build-$(date +%Y%m%d)"
+CSDK_VER="master"  # Build from master
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: `AWS_ENCRYPTION_SDK_C_VER`

File: codebuild/osx.sh [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set +ex
+
+OS=$(uname -s)
+CPP_VER="1.7.163"  #github versioned branch
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: `AWS_SDK_CPP_VER`

File: codebuild/osx.sh [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+BUILDROOT="/var/tmp/build-$(date +%s)"
+OPENSSLVER="openssl@1.1" #note this is a brew label, not an exact version reference
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: NP: maintain same variable style `OPENSSL_VER`

File: codebuild/osx.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    xcodebuild -target ALL_BUILD 
+    xcodebuild -target install
+
+}
+
+build_csdk(){
+    echo "Building csdk"
+    cd ${BUILDROOT}
+    git clone -b ${CSDK_VER} https://github.com/aws/aws-encryption-sdk-c.git 
+    mkdir -p ${BUILDROOT}/build-csdk || true
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: `build-aws-encryption-sdk-c`

File: README.md [hunk 8]
```diff
@@ -161,7 +161,7 @@ Once it is set up, run the following:
 takes much longer than the KMS-only build. For these reasons, we recommend doing a source
 build of the AWS SDK for C++ yourself, as we will demonstrate below.)
 
-Start in the directory where you want to do your build.
+Start in the directory where you want to do your build, or use the script in ./codebuild/osx.sh.
```
[the remark this answers is not in the document]
Comment (dougch) [reply README.md]: Removed this comment, for a future rewrite of the README(s).  My intent was to have the same script used by codebuild and humans, if possible.

File: README.md [hunk 9]
```diff
@@ -194,8 +194,8 @@ set the directory manually in our build.
 
     git clone https://github.com/aws/aws-encryption-sdk-c.git
     mkdir build-aws-encryption-sdk-c && cd build-aws-encryption-sdk-c
-    cmake -DBUILD_SHARED_LIBS=ON -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl\@1.1 ../aws-encryption-sdk-c
-    make && sudo make install ; cd ..
+    cmake -G Xcode -DBUILD_SHARED_LIBS=ON -DOPENSSL_ROOT_DIR="/usr/local/opt/openssl@1.1" ../aws-encryption-sdk-c 
+    xcodebuild -target install
```
[the remark this answers is not in the document]
Comment (dougch) [reply README.md]: addressed in next commit

File: codebuild/osx.sh (hunk 2 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: codebuild/osx.sh (hunk 4 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: codebuild/osx.sh (hunk 6 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: codebuild/osx.sh (hunk 7 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: codebuild/osx.sh (hunk 5 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: codebuild/osx.sh (hunk 3 above)
Comment (dougch) [reply codebuild/osx.sh]: addressed in next commit

File: README.md (hunk 1 above)
Comment (dougch) [reply README.md]: addressed in next commit

File: codebuild/osx.sh [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+OS=$(uname -s)
+AWS_SDK_CPP_VER="1.7.163"  #github versioned branch
+BUILDROOT="/var/tmp/build-$(date +%s)"
+OPENSSL_VER="openssl@1.1" #note this is a brew label, not an exact version reference
+BUILDROOT="/var/tmp/build-$(date +%Y%m%d)"
+AWS_ENCRYPTION_SDK_C_VER="master"  # Build from master
+
+deps(){
+  if [ $(which brew|grep -c 'brew') -lt 1 ]; then
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]: Could we add a log statement stating building dependencies.

File: codebuild/osx.sh [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+OS=$(uname -s)
+AWS_SDK_CPP_VER="1.7.163"  #github versioned branch
+BUILDROOT="/var/tmp/build-$(date +%s)"
+OPENSSL_VER="openssl@1.1" #note this is a brew label, not an exact version reference
+BUILDROOT="/var/tmp/build-$(date +%Y%m%d)"
+AWS_ENCRYPTION_SDK_C_VER="master"  # Build from master
+
+deps(){
+  echo "Installing dependancies"
```
Comment (ttjsu-aws) [inline codebuild/osx.sh]:
```suggestion
  echo "Installing dependencies" 
```

**Pull request merged**